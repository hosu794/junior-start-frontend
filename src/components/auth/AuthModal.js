import React, {Component} from "react";
import SignInForm from "./SignInForm";
import {Divider} from "@material-ui/core";
import SignUpForm from "./SignUpForm";
import {CustomPaper} from "../../styles/paperStyles";

class AuthModal extends Component {

    render() {
        return (
            <CustomPaper tabIndex={-1}>
                <SignInForm/>
                <Divider orientation="vertical" flexItem/>
                <SignUpForm/>
            </CustomPaper>
        );
    }
}

export default AuthModal;