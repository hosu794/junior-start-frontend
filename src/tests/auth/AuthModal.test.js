import { render } from '@testing-library/react'
import renderer from 'react-test-renderer';
import React from 'react'
import AuthModal from '../../components/auth/AuthModal'
import { MemoryRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

const mockStore = configureStore([]);

let store = mockStore({
    authentication: {
        loading: false,
        error: true,
    },
});

const TestingComponent = () => (
    <MemoryRouter>
        <Provider store={store}>
            <AuthModal />
        </Provider>
    </MemoryRouter>
);

it('should render correctly', async () => {

    const tree = renderer.create(<TestingComponent />).toJSON();
    expect(tree).toMatchSnapshot();

})

it('should render SignIn component', () => {
    const { getByText } = render(<TestingComponent />)
    expect(getByText('Zaloguj się')).toBeInTheDocument()
})

it('should render SignUp component', () => {
    const { getByText } = render(<TestingComponent />)
    expect(getByText('Zarejestruj się')).toBeInTheDocument()
})



