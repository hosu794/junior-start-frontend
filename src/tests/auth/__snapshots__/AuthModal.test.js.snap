// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="MuiPaper-root sc-eCApGN iwCFWv MuiPaper-elevation1 MuiPaper-rounded"
  tabIndex={-1}
>
  <div
    className="sc-bdnylx hNwEPP"
  >
    <h3>
      Zaloguj się
    </h3>
    <form
      action="#"
      className="sc-gtssRu gWMPic"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          Adres email
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="email"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Adres email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          Hasło
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Hasło
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormGroup-root MuiFormGroup-row"
      >
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                data-indeterminate={false}
                name="staySignedIn"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Nie wylogowuj mnie
          </span>
        </label>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Zaloguj
        </span>
      </button>
      <p
        className="sc-hKFyIo ilhKaQ"
      >
        Nieprawidłowy login lub hasło. Spróbuj ponownie.
      </p>
    </form>
    <div
      className="social-login"
    >
      <a
        className="btn btn-block social-btn google"
        href="https://junior-start-backend.herokuapp.com/oauth2/authorize/google?redirect_uri=http://localhost:3000/oauth2/redirect"
      >
        <img
          alt="Google"
          src="google-logo.png"
        />
         Log in with Google
      </a>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
  />
  <div
    className="sc-bdnylx hNwEPP"
    id="form"
  >
    <h3>
      Zarejestruj się
    </h3>
    <form
      action="#"
      className="sc-gtssRu gWMPic"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          Nazwa użytkownika
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Nazwa użytkownika
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          E-mail
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="email"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                E-mail
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          Hasło
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Hasło
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          Potwierdź hasło
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            name="passwordConfirm"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Potwierdź hasło
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormGroup-root MuiFormGroup-row"
      >
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                data-indeterminate={false}
                name="acceptedTerms"
                onChange={[Function]}
                type="checkbox"
                value={false}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Akceptuję regulamin
          </span>
        </label>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Zarejestruj
        </span>
      </button>
    </form>
    <div
      className="social-signup"
    >
      <a
        className="btn btn-block social-btn google"
        href="https://junior-start-backend.herokuapp.com/oauth2/authorize/google?redirect_uri=http://localhost:3000/oauth2/redirect"
      >
        <img
          alt="Google"
          src="google-logo.png"
        />
         Sign up with Google
      </a>
    </div>
  </div>
</div>
`;
